marconi-tutorial/src/Marconi/Tutorial/Env.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Marconi.Tutorial.Env where
+
+import Cardano.BM.Trace (Trace)
+import Control.Concurrent (MVar)
+import Control.Concurrent.MVar (newEmptyMVar)
+import Control.Lens (makeLenses)
+import Data.IORef (IORef, newIORef)
+import Data.Text (Text)
+import Marconi.Tutorial.CLI qualified as CLI
+import Marconi.Tutorial.Indexers.AddressCount (AddressCountIndexer)
+
+data Env = Env
+  { _envIndexers :: IndexersEnv
+  -- ^ Access the Tutorial indexers for querying purposes.
+  , _envCliArgs :: !CLI.Options
+  , _envStdoutTrace :: !(Trace IO Text)
+  }
+
+-- | Should contain all the indexers required by the tutorial.
+newtype IndexersEnv = IndexersEnv
+  { _addressCountIndexerEnv :: AddressCountIndexerEnv
+  }
+
+newtype AddressCountIndexerEnv = AddressCountIndexerEnv
+  { _addressCountIndexerVar :: IORef (MVar AddressCountIndexer)
+  }
```

Comment:
I do understand the idea of providing a taste of how we would create an adequate architecture for a larger set of indexers, but with only one indexer, it introduces a lot of indirections.

Reply:
True. We have 2 options:

1- Keep this package as simple as possible as just used for explaining how to write indexer. Then, add a 2nd package called `marconi-starter` which provides the adequate architecture for starting a Marconi project.
2- Have `marconi-tutorial` be both a tutorial + project scaffold.

Reply:
I like the `marconi-starter` idea, as using several indexers may lead to other complications, as the need to use a `Coordinator`.